---
title: "ACR UkBioBank Gout Summary Report"
author: "Murray Cadzow"
date: '`r format(Sys.Date())`'
output:
  pdf_document: 
    toc: yes
    includes:
      in_header: header.tex
---

Exclusion criteria:

Gout:
anyone with renal disease
identified though ULT alone: has leukemia or lymphoma

any control:
anyone with renal disease
classified in any way as having gout
taking gout medication or NSAIDs or cortico steroidal prednisone


genotypes:
genotyping qc fail
gender mismatch

removed related individuals

```{r load data,echo = FALSE, message=FALSE, warning=FALSE}
library(dplyr)
library(ggplot2)
library(pander)
# read in fam file for sample filtered plink files
data_dir <- "/media/xsan/staff_groups/merrimanlab/Merriman_Documents/Murray/ukbiobank_util/data/"
scratch_dir <- "/media/xsan/scratch/merrimanlab/murray/working_dir/UkBio/"
control_cond <- 'controls' 

load(paste0(data_dir,"ukbiobank_genotyped2016-04-26.RData"))
genotyped$waist_height_ratio <- genotyped$f.48.0.0 / genotyped$f.50.0.0

source("create_kottgen_table1.R")
source('read_fams.R')

gout_cond <- 'all'
all_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(all_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_all <- NA
genotyped[genotyped$f.eid %in% all_fam$IID & 
                  genotyped$f.eid %in% all_fam[which(all_fam$AFF==1 | all_fam$AFF == 2),"IID"], "fam_all"] <-1

genotyped$fam <- NA
genotyped[genotyped$f.eid %in% all_fam$IID,"fam"] <- 1

gout_cond <- 'self'
self_fam  <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(self_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_self <- NA
genotyped[genotyped$f.eid %in% self_fam$IID & 
                  genotyped$f.eid %in% self_fam[which(self_fam$AFF==1 | self_fam$AFF == 2),"IID"], "fam_self"] <-1

gout_cond <- 'self_ult'
self_ult_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(self_ult_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_self_ult <- NA
genotyped[genotyped$f.eid %in% self_ult_fam$IID & 
                  genotyped$f.eid %in% self_ult_fam[which(self_ult_fam$AFF==1 | self_ult_fam$AFF == 2),"IID"], "fam_self_ult"] <-1

gout_cond <- 'ult'
ult_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(ult_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_ult <- NA
genotyped[genotyped$f.eid %in% ult_fam$IID & 
                  genotyped$f.eid %in% ult_fam[which(ult_fam$AFF==1 | ult_fam$AFF == 2),"IID"], "fam_ult"] <-1

gout_cond <- 'winnard'
winnard_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(winnard_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_winnard <- NA
genotyped[genotyped$f.eid %in% winnard_fam$IID & 
                  genotyped$f.eid %in% winnard_fam[which(winnard_fam$AFF==1 | winnard_fam$AFF == 2),"IID"], "fam_winnard"] <-1

gout_cond <- 'hosp'
hosp_fam  <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(hosp_fam)  <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_hosp <- NA
genotyped[genotyped$f.eid %in% hosp_fam$IID & 
                  genotyped$f.eid %in% hosp_fam[which(hosp_fam$AFF==1 | hosp_fam$AFF == 2),"IID"], "fam_hosp"] <-1
```


![Genotyped UK Biobank Gout Patients](genotyped_venn.png)


# Genotyped - Controls unfiltered diuretics

There are a total of ```r as.character(genotyped %>% select(fam)  %>% filter(fam == 1) %>% tally())``` people with genotypes

The GWAS were limited to only British, Irish, and "any other white background" ethnicities because there were only ```r as.data.frame(genotyped  %>% filter(!is.na(goutaff) & !(f.21000.0.0 %in% c(1001,1002,1003)) & f.eid %in% all_fam$IID)  %>%  select(f.21000.0.0, goutaff)  %>% group_by(goutaff, f.21000.0.0)  %>%  tally()  %>% summarise(sum(n)))[2,2]``` gout cases from all other ethnicities with genotypes




## All


```{r all_gwas_stats, echo=FALSE}
gout_cond <- 'all'
all <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
all_drug_tally <- all[[2]]
all_summary_table <-  all[[1]]

```
For the GWAS based on all gout criteria there were ```r all_summary_table['n',"Gout"]``` cases and ```r all_summary_table['n',"Control"]``` controls

There were:
  
- ```r as.character(all_drug_tally$diuretics_cases)``` cases on diuretics, ```r all_drug_tally$diuretics_controls``` were controls

- ```r all_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r all_drug_tally$allopurinol_controls``` were controls

- ```r all_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r all_drug_tally$febuxostat_controls``` were controls

- ```r all_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r all_drug_tally$sulphinpyrazone_controls``` were controls

- ```r all_drug_tally$colchicine_cases``` people taking cochicine, ```r all_drug_tally$colchicine_controls``` were controls

Age summary

```{r all_age_summary, echo = FALSE}
as.data.frame(genotyped  %>% 
  select(fam_all, goutaff, f.21003.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21003.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21003.0.0), sd = sd(f.21003.0.0), min = min(f.21003.0.0),median = median(f.21003.0.0), max = max(f.21003.0.0) ), row.names = NULL)
```

Sex summary

```{r all_gwas_sex_plot, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_all) %>% 
  filter(!is.na(goutaff), fam_all == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("All Sex Distribution")
```

```{r all_gwas_bmi_plot, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_all, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("All Gout Criteria")
```

```{r all_gwas_wh_plot, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_all, goutaff, waist_height_ratio , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(waist_height_ratio)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = waist_height_ratio)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("Waist to Height Ratio") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("All Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval = FALSE}
as.data.frame(genotyped  %>% 
  select(fam_all, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```


### All summary table



```{r all_summary_table, echo = FALSE, results= 'asis'}
pandoc.table(all_summary_table, caption = 'Summary Table - All')
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results='asis'}
gout_cond <- 'all'
all_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(all_kot[['sum_table']]) <- NULL
pandoc.table(as.data.frame(all_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")], row.names = NULL), caption = 'Kottgen table 1 SNPs - All')

```


### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results='asis'}
gout_cond <- 'all'
all_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(all_kot[['sum_table']]) <- NULL
pandoc.table(as.data.frame(all_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")], row.names = NULL), caption = 'Kottgen table 1 SNPs - All')

```



![Gout all - adjusted: age sex waist and waist to height](controls_all_adjusted.png)

\pagebreak

\newpage
## Winnard

```{r winnard_gwas_stats, echo = FALSE}
gout_cond <- 'winnard'
winnard <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
winnard_drug_tally <- winnard[[2]]
winnard_summary_table <-  winnard[[1]]

```


For the GWAS based on the Winnard et al gout criteria there were ```r winnard_summary_table['n',"Gout"]``` cases and ```r as.character(winnard_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(winnard_drug_tally$diuretics_cases)``` cases on diuretics, ```r winnard_drug_tally$diuretics_controls``` were controls

- ```r winnard_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r winnard_drug_tally$allopurinol_controls``` were controls

- ```r winnard_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r winnard_drug_tally$febuxostat_controls``` were controls

- ```r winnard_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r winnard_drug_tally$sulphinpyrazone_controls``` were controls

- ```r winnard_drug_tally$colchicine_cases``` people taking cochicine, ```r winnard_drug_tally$colchicine_controls``` were controls


```{r winnard_gwas_sex_plot, echo = FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_winnard) %>% 
  filter(!is.na(goutaff), fam_winnard == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Winnard Sex Distribution")
```

```{r winnard_gwas_bmi_plot, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_winnard, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_winnard == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Winnard Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_winnard, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_winnard == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc,sex) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```


### Winnard summary table

```{r winnard_summary_table, echo = FALSE, results = 'asis'}
pandoc.table(winnard_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'winnard'
winnard_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(winnard_kot[['sum_table']]) <- NULL
pandoc.table(winnard_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'winnard'
winnard_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(winnard_kot[['sum_table']]) <- NULL
pandoc.table(winnard_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

![Gout winnard - adjusted: age sex waist and waist to height](controls_winnard_adjusted.png)


\pagebreak

\newpage
## Hospital

```{r hosp_gwas_stats, echo = FALSE}
gout_cond <- 'hosp'
hosp <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
hosp_drug_tally <- hosp[[2]]
hosp_summary_table <-  hosp[[1]]

```


For the GWAS based on hospital diagnosis of gout criteria there were ```r hosp_summary_table['n',"Gout"]``` cases and ```r as.character(hosp_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(hosp_drug_tally$diuretics_cases)``` cases on diuretics, ```r hosp_drug_tally$diuretics_controls``` were controls

- ```r hosp_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r hosp_drug_tally$allopurinol_controls``` were controls

- ```r hosp_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r hosp_drug_tally$febuxostat_controls``` were controls

- ```r hosp_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r hosp_drug_tally$sulphinpyrazone_controls``` were controls

- ```r hosp_drug_tally$colchicine_cases``` people taking cochicine, ```r hosp_drug_tally$colchicine_controls``` were controls


```{r hosp_gwas_sex_plot, echo = FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_hosp) %>% 
  filter(!is.na(goutaff), fam_hosp == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Hospital Sex Distribution")
```

```{r hosp_gwas_bmi_plot, echo=FALSE, message=FALSE, warning=FALSE, eval = FALSE}
genotyped  %>% 
  select(fam_hosp, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_hosp == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Hospital Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_hosp, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_hosp == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Hospital summary table

```{r hosp_summary_table, echo = FALSE, results= 'asis'}
pandoc.table(hosp_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'hosp'
hosp_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(hosp_kot[['sum_table']]) <- NULL
pandoc.table(hosp_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```


### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'hosp'
hosp_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(hosp_kot[['sum_table']]) <- NULL
pandoc.table(hosp_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

![Gout hospital - adjusted: age sex waist and waist to height](controls_hosp_adjusted.png)


\pagebreak

\newpage
## Self Reported

```{r self_gwas_stats, echo=FALSE}
gout_cond <- 'self'
self <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
self_drug_tally <- self[[2]]
self_summary_table <-  self[[1]]

```



For the GWAS based on self reported gout criteria there were ```r self_summary_table['n',"Gout"]``` cases and ```r as.character(self_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(self_drug_tally$diuretics_cases)``` cases on diuretics, ```r self_drug_tally$diuretics_controls``` were controls

- ```r self_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r self_drug_tally$allopurinol_controls``` were controls

- ```r self_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r self_drug_tally$febuxostat_controls``` were controls

- ```r self_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r self_drug_tally$sulphinpyrazone_controls``` were controls

- ```r self_drug_tally$colchicine_cases``` people taking cochicine, ```r self_drug_tally$colchicine_controls``` were controls



```{r self_gwas_sex_plot, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_self) %>% 
  filter(!is.na(goutaff), fam_self == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Self Report Sex Distribution")
```

```{r self_gwas_bmi_plot, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_self, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Self Report Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_self, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Self report summary table

```{r self_summary_table, echo = FALSE, results= 'asis'}
pandoc.table(self_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self'
self_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(self_kot[['sum_table']]) <- NULL
pandoc.table(self_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```



### Kottgen table 1 SNPs - age sex waist and waist to height adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self'
self_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(self_kot[['sum_table']]) <- NULL
pandoc.table(self_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout self - adjusted: age sex waist and waist to height](controls_self_adjusted.png)

\pagebreak


\newpage
## ULT
```{r ult_gwas_stats1, echo = FALSE}
gout_cond <- 'ult'
ult <- read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
ult_drug_tally <- ult[[2]]
ult_summary_table <-  ult[[1]]
```

For the GWAS based on ULT gout criteria there were ```r ult_summary_table['n',"Gout"]``` cases and ```r as.character(ult_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(ult_drug_tally$diuretics_cases)``` cases on diuretics, ```r ult_drug_tally$diuretics_controls``` were controls

- ```r ult_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r ult_drug_tally$allopurinol_controls``` were controls

- ```r ult_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r ult_drug_tally$febuxostat_controls``` were controls

- ```r ult_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r ult_drug_tally$sulphinpyrazone_controls``` were controls

- ```r ult_drug_tally$colchicine_cases``` people taking cochicine, ```r ult_drug_tally$colchicine_controls``` were controls



```{r ult_gwas_sex_plot, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_ult) %>% 
  filter(!is.na(goutaff), fam_ult == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("ULT Sex Distribution")
```

```{r ult_gwas_bmi_plot, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("ULT Gout Criteria")
```

BMI Summary:
  
```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
                select(fam_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
                filter(fam_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
                mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
                group_by(cc) %>% 
                summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### ULT summary table

```{r ult_summary_table, echo = FALSE, results = 'asis'}
pandoc.table(ult_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'ult'
ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(ult_kot[['sum_table']]) <- NULL
pandoc.table(ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```


### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'ult'
ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(ult_kot[['sum_table']]) <- NULL
pandoc.table(ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout ult - adjusted: age sex waist and waist to height](controls_ult_adjusted.png)

\newpage
## Self Reported and ULT

```{r self_ult_gwas_stats, echo = FALSE}
gout_cond <- 'self_ult'
self_ult <- read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
self_ult_drug_tally <- self_ult[[2]]
self_ult_summary_table <-  self_ult[[1]]
```

For the GWAS based on self report and ULT gout criteria there were ```r self_ult_summary_table['n',"Gout"]``` cases and ```r as.character(self_ult_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(self_ult_drug_tally$diuretics_cases)``` cases on diuretics, ```r self_ult_drug_tally$diuretics_controls``` were controls

- ```r self_ult_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r self_ult_drug_tally$allopurinol_controls``` were controls

- ```r self_ult_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r self_ult_drug_tally$febuxostat_controls``` were controls

- ```r self_ult_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r self_ult_drug_tally$sulphinpyrazone_controls``` were controls

- ```r self_ult_drug_tally$colchicine_cases``` people taking cochicine, ```r self_ult_drug_tally$colchicine_controls``` were controls



```{r self_ult_gwas_sex_plot, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_self_ult) %>% 
  filter(!is.na(goutaff), fam_self_ult == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Self Report and ULT Sex Distribution")
```

```{r self_ult_gwas_bmi_plot, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_self_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Self Report and ULT Gout Criteria")
```

BMI Summary:
  
```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
                select(fam_self_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
                filter(fam_self_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
                mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
                group_by(cc) %>% 
                summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Self report and ULT summary table

```{r self_ult_summary_table, echo = FALSE, results = 'asis'}
pandoc.table(self_ult_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self_ult'
self_ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(self_ult_kot[['sum_table']]) <- NULL
pandoc.table(self_ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self_ult'
self_ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(self_ult_kot[['sum_table']]) <- NULL
pandoc.table(self_ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout self and ult - adjusted: age sex waist and waist to height](controls_self_ult_adjusted.png)

\newpage
## Combined tables

\newpage
\blandscape



```{r, echo = FALSE, results = 'asis'}
kot_t1 <- read.csv("kottgen_table1.csv", header = TRUE, stringsAsFactors = FALSE)
kot_t1_gout <- kot_t1[,c("SNP","CHR","BP","grail_gene","OR","gout_pvalue", "A1", "A2")]
names(kot_t1_gout) <- c("SNP","CHR","BP","grail_gene","OR","P", "A1","A2")
kot_t1_gout$lower95 <- NA
kot_t1_gout$upper95 <- NA
kot_t1_gout$SE <- NA
```

```{r, echo = FALSE, message=FALSE}
library(RMySQL)
library(data.table)
library(dplyr)
drv = dbDriver("MySQL")
db = dbConnect(drv, default.file = '~/.my.cnf', dbname="merrimanDW_test")

ukvariant <- dbGetQuery(db, paste('select * from dimVariant where snp in (', paste(paste0("'",kot_t1$SNP,"'"), collapse = ','),")"))
```



```{r, echo=FALSE, results = 'asis'}
kottgen_table1 <- list(all =merge(all_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), winnard = merge(winnard_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), hosp = merge(hosp_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), self = merge(self_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE),ult = merge(ult_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), self_ult = merge(self_ult_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), kottgen_gout = kot_t1_gout) 
for (name in names(kottgen_table1)){
  kottgen_table1[[name]]$gout <-  name
}



kt1 <- do.call(rbind, kottgen_table1)
kt1 <-kt1[order(kt1$SNP), ! colnames(kt1) %in% c('sort')]
row.names(kt1) <- NULL
#pandoc.table(kt1[,c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P","gout")], caption = 'Combined Kottgen Table 1 All Gout Classifications' )
pandoc.table(kt1[,c('SNP','CHR','BP',"A1","A2", "grail_gene","OR","P","gout")], caption = 'Combined Kottgen Table 1 All Gout Classifications - adjusted' )
```

\elandscape

\newpage

```{r, echo = FALSE, fig.align='centre', fig.width= 8, fig.height=8}
ggplot(data = kt1, aes(x = gout, y = OR)) + geom_point() + facet_wrap(~SNP, scales = 'free_y') + geom_linerange(aes(x = gout, ymin = lower95, ymax = upper95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90)) 

```


```{r, echo = FALSE, fig.width=8, fig.height=8, eval = FALSE}
dat_kottgen_table1 <- list(all =all_kot[['dat']], winnard = winnard_kot[['dat']], hosp = hosp_kot[['dat']], self = self_kot[['dat']],ult = ult_kot[['dat']], self_ult = self_ult_kot[['dat']]) 
for (name in names(dat_kottgen_table1)){
  dat_kottgen_table1[[name]]$gout <-  name
}
dat_kt1 <- do.call(rbind, dat_kottgen_table1)
dat_kt1 <-dat_kt1[order(dat_kt1$SNP), ! colnames(dat_kt1) %in% c('sort')]
row.names(dat_kt1) <- NULL
#pandoc.table(dat_kt1[,], caption = 'Combined Kottgen Table 1 All Gout Classifications' )

ggplot(data = dat_kt1, aes(x = gout, y = OR)) + geom_point() + facet_wrap(~SNP, scales = 'free_y') + geom_linerange(aes(x = gout, ymin = L95, ymax = L95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90)) + stat_summary(fun.y=mean, geom="point", shape=18,size=3, color="red") 
``` 






```{r load data no diuretics,echo = FALSE, message=FALSE, warning=FALSE}
control_cond <- 'controls_no_diuretics'

load(paste0(data_dir,"ukbiobank_genotyped2016-04-26.RData"))
genotyped$waist_height_ratio <- genotyped$f.48.0.0 / genotyped$f.50.0.0

source("create_kottgen_table1.R")
source('read_fams.R')

gout_cond <- 'all'
all_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(all_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_all <- NA
genotyped[genotyped$f.eid %in% all_fam$IID & 
                  genotyped$f.eid %in% all_fam[which(all_fam$AFF==1 | all_fam$AFF == 2),"IID"], "fam_all"] <-1

genotyped$fam <- NA
genotyped[genotyped$f.eid %in% all_fam$IID,"fam"] <- 1

gout_cond <- 'self'
self_fam  <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(self_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_self <- NA
genotyped[genotyped$f.eid %in% self_fam$IID & 
                  genotyped$f.eid %in% self_fam[which(self_fam$AFF==1 | self_fam$AFF == 2),"IID"], "fam_self"] <-1

gout_cond <- 'self_ult'
self_ult_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(self_ult_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_self_ult <- NA
genotyped[genotyped$f.eid %in% self_ult_fam$IID & 
                  genotyped$f.eid %in% self_ult_fam[which(self_ult_fam$AFF==1 | self_ult_fam$AFF == 2),"IID"], "fam_self_ult"] <-1

gout_cond <- 'winnard'
winnard_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(winnard_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_winnard <- NA
genotyped[genotyped$f.eid %in% winnard_fam$IID & 
                  genotyped$f.eid %in% winnard_fam[which(winnard_fam$AFF==1 | winnard_fam$AFF == 2),"IID"], "fam_winnard"] <-1

gout_cond <- 'ult'
ult_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(ult_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_ult <- NA
genotyped[genotyped$f.eid %in% ult_fam$IID & 
                  genotyped$f.eid %in% ult_fam[which(ult_fam$AFF==1 | ult_fam$AFF == 2),"IID"], "fam_ult"] <-1


gout_cond <- 'hosp'
hosp_fam  <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(hosp_fam)  <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_hosp <- NA
genotyped[genotyped$f.eid %in% hosp_fam$IID & 
                  genotyped$f.eid %in% hosp_fam[which(hosp_fam$AFF==1 | hosp_fam$AFF == 2),"IID"], "fam_hosp"] <-1
```

\newpage
# Genotyped - Controls no diuretics

There are a total of ```r as.character(genotyped %>% select(fam)  %>% filter(fam == 1) %>% tally())``` people with genotypes

The GWAS were limited to only British, Irish, and "any other white background" ethnicities because there were only ```r as.data.frame(genotyped  %>% filter(!is.na(goutaff) & !(f.21000.0.0 %in% c(1001,1002,1003)) & f.eid %in% all_fam$IID)  %>%  select(f.21000.0.0, goutaff)  %>% group_by(goutaff, f.21000.0.0)  %>%  tally()  %>% summarise(sum(n)))[2,2]``` gout cases from all other ethnicities with genotypes

## All


```{r all_gwas_stats2, echo=FALSE}
gout_cond <- 'all'
all <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
all_drug_tally <- all[[2]]
all_summary_table <-  all[[1]]

```
For the GWAS based on all gout criteria there were ```r all_summary_table['n',"Gout"]``` cases and ```r all_summary_table['n',"Control"]``` controls

There were:
  
- ```r as.character(all_drug_tally$diuretics_cases)``` cases on diuretics, ```r all_drug_tally$diuretics_controls``` were controls

- ```r all_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r all_drug_tally$allopurinol_controls``` were controls

- ```r all_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r all_drug_tally$febuxostat_controls``` were controls

- ```r all_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r all_drug_tally$sulphinpyrazone_controls``` were controls

- ```r all_drug_tally$colchicine_cases``` people taking cochicine, ```r all_drug_tally$colchicine_controls``` were controls

Age summary

```{r all_age_summary2, echo = FALSE}
as.data.frame(genotyped  %>% 
  select(fam_all, goutaff, f.21003.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21003.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21003.0.0), sd = sd(f.21003.0.0), min = min(f.21003.0.0),median = median(f.21003.0.0), max = max(f.21003.0.0) ), row.names = NULL)
```

Sex summary

```{r all_gwas_sex_plot2, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_all) %>% 
  filter(!is.na(goutaff), fam_all == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("All Sex Distribution")
```

```{r all_gwas_bmi_plot2, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_all, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("All Gout Criteria")
```

```{r all_gwas_wh_plot2, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_all, goutaff, waist_height_ratio , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(waist_height_ratio)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = waist_height_ratio)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("Waist to Height Ratio") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("All Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval = FALSE}
as.data.frame(genotyped  %>% 
  select(fam_all, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```


### All summary table



```{r all_summary_table2, echo = FALSE, results= 'asis'}
pandoc.table(all_summary_table, caption = 'Summary Table - All')
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results='asis'}
gout_cond <- 'all'
all_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(all_kot[['sum_table']]) <- NULL
pandoc.table(as.data.frame(all_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")], row.names = NULL), caption = 'Kottgen table 1 SNPs - All')

```


### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results='asis'}
gout_cond <- 'all'
all_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(all_kot[['sum_table']]) <- NULL
pandoc.table(as.data.frame(all_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")], row.names = NULL), caption = 'Kottgen table 1 SNPs - All')

```

![Gout all - adjusted: age sex waist and waist to height](controls_no_diuretics_all_adjusted.png)


\pagebreak

\newpage
## Winnard

```{r winnard_gwas_stats2, echo = FALSE}
gout_cond <- 'winnard'
winnard <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
winnard_drug_tally <- winnard[[2]]
winnard_summary_table <-  winnard[[1]]

```


For the GWAS based on the Winnard et al gout criteria there were ```r winnard_summary_table['n',"Gout"]``` cases and ```r as.character(winnard_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(winnard_drug_tally$diuretics_cases)``` cases on diuretics, ```r winnard_drug_tally$diuretics_controls``` were controls

- ```r winnard_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r winnard_drug_tally$allopurinol_controls``` were controls

- ```r winnard_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r winnard_drug_tally$febuxostat_controls``` were controls

- ```r winnard_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r winnard_drug_tally$sulphinpyrazone_controls``` were controls

- ```r winnard_drug_tally$colchicine_cases``` people taking cochicine, ```r winnard_drug_tally$colchicine_controls``` were controls


```{r winnard_gwas_sex_plot2, echo = FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_winnard) %>% 
  filter(!is.na(goutaff), fam_winnard == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Winnard Sex Distribution")
```

```{r winnard_gwas_bmi_plot2, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_winnard, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_winnard == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Winnard Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_winnard, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_winnard == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc,sex) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```


### Winnard summary table

```{r winnard_summary_table2, echo = FALSE, results = 'asis'}
pandoc.table(winnard_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'winnard'
winnard_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(winnard_kot[['sum_table']]) <- NULL
pandoc.table(winnard_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'winnard'
winnard_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(winnard_kot[['sum_table']]) <- NULL
pandoc.table(winnard_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

![Gout winnard - adjusted: age sex waist and waist to height](controls_no_diuretics_winnard_adjusted.png)

\pagebreak

\newpage
## Hospital

```{r hosp_gwas_stats2, echo = FALSE}
gout_cond <- 'hosp'
hosp <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
hosp_drug_tally <- hosp[[2]]
hosp_summary_table <-  hosp[[1]]

```


For the GWAS based on hospital diagnosis of gout criteria there were ```r hosp_summary_table['n',"Gout"]``` cases and ```r as.character(hosp_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(hosp_drug_tally$diuretics_cases)``` cases on diuretics, ```r hosp_drug_tally$diuretics_controls``` were controls

- ```r hosp_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r hosp_drug_tally$allopurinol_controls``` were controls

- ```r hosp_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r hosp_drug_tally$febuxostat_controls``` were controls

- ```r hosp_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r hosp_drug_tally$sulphinpyrazone_controls``` were controls

- ```r hosp_drug_tally$colchicine_cases``` people taking cochicine, ```r hosp_drug_tally$colchicine_controls``` were controls


```{r hosp_gwas_sex_plot2, echo = FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_hosp) %>% 
  filter(!is.na(goutaff), fam_hosp == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Hospital Sex Distribution")
```

```{r hosp_gwas_bmi_plot2, echo=FALSE, message=FALSE, warning=FALSE, eval = FALSE}
genotyped  %>% 
  select(fam_hosp, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_hosp == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Hospital Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_hosp, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_hosp == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Hospital summary table

```{r hosp_summary_table2, echo = FALSE, results= 'asis'}
pandoc.table(hosp_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'hosp'
hosp_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(hosp_kot[['sum_table']]) <- NULL
pandoc.table(hosp_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'hosp'
hosp_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(hosp_kot[['sum_table']]) <- NULL
pandoc.table(hosp_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

![Gout hospital - adjusted: age sex waist and waist to height](controls_no_diuretics_hosp_adjusted.png)

\pagebreak

\newpage
## Self Reported

```{r self_gwas_stats2, echo=FALSE}
gout_cond <- 'self'
self <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
self_drug_tally <- self[[2]]
self_summary_table <-  self[[1]]

```



For the GWAS based on self reported gout criteria there were ```r self_summary_table['n',"Gout"]``` cases and ```r as.character(self_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(self_drug_tally$diuretics_cases)``` cases on diuretics, ```r self_drug_tally$diuretics_controls``` were controls

- ```r self_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r self_drug_tally$allopurinol_controls``` were controls

- ```r self_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r self_drug_tally$febuxostat_controls``` were controls

- ```r self_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r self_drug_tally$sulphinpyrazone_controls``` were controls

- ```r self_drug_tally$colchicine_cases``` people taking cochicine, ```r self_drug_tally$colchicine_controls``` were controls



```{r self_gwas_sex_plot2, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_self) %>% 
  filter(!is.na(goutaff), fam_self == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Self Report Sex Distribution")
```

```{r self_gwas_bmi_plot2, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_self, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Self Report Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_self, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Self report summary table

```{r self_summary_table2, echo = FALSE, results= 'asis'}
pandoc.table(self_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self'
self_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(self_kot[['sum_table']]) <- NULL
pandoc.table(self_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

### Kottgen table 1 SNPs - age sex waist and waist to height adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self'
self_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(self_kot[['sum_table']]) <- NULL
pandoc.table(self_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout self - adjusted: age sex waist and waist to height](controls_no_diuretics_self_adjusted.png)

\pagebreak

\newpage
## ULT
```{r ult_gwas_stats2, echo = FALSE}
gout_cond <- 'ult'
ult <- read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
ult_drug_tally <- ult[[2]]
ult_summary_table <-  ult[[1]]
```

For the GWAS based on ULT gout criteria there were ```r ult_summary_table['n',"Gout"]``` cases and ```r as.character(ult_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(ult_drug_tally$diuretics_cases)``` cases on diuretics, ```r ult_drug_tally$diuretics_controls``` were controls

- ```r ult_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r ult_drug_tally$allopurinol_controls``` were controls

- ```r ult_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r ult_drug_tally$febuxostat_controls``` were controls

- ```r ult_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r ult_drug_tally$sulphinpyrazone_controls``` were controls

- ```r ult_drug_tally$colchicine_cases``` people taking cochicine, ```r ult_drug_tally$colchicine_controls``` were controls



```{r ult_gwas_sex_plot2, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_ult) %>% 
  filter(!is.na(goutaff), fam_ult == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("ULT Sex Distribution")
```

```{r ult_gwas_bmi_plot2, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("ULT Gout Criteria")
```

BMI Summary:
  
```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
                select(fam_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
                filter(fam_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
                mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
                group_by(cc) %>% 
                summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### ULT summary table

```{r ult_summary_table2, echo = FALSE, results = 'asis'}
pandoc.table(ult_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'ult'
ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(ult_kot[['sum_table']]) <- NULL
pandoc.table(ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```


### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'ult'
ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(ult_kot[['sum_table']]) <- NULL
pandoc.table(ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout ult - adjusted: age sex waist and waist to height](controls_no_diuretics_ult_adjusted.png)

\newpage
## Self Reported and ULT

```{r self_ult_gwas_stats2, echo = FALSE}
gout_cond <- 'self_ult'
self_ult <- read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
self_ult_drug_tally <- self_ult[[2]]
self_ult_summary_table <-  self_ult[[1]]
```

For the GWAS based on self report and ULT gout criteria there were ```r self_ult_summary_table['n',"Gout"]``` cases and ```r as.character(self_ult_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(self_ult_drug_tally$diuretics_cases)``` cases on diuretics, ```r self_ult_drug_tally$diuretics_controls``` were controls

- ```r self_ult_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r self_ult_drug_tally$allopurinol_controls``` were controls

- ```r self_ult_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r self_ult_drug_tally$febuxostat_controls``` were controls

- ```r self_ult_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r self_ult_drug_tally$sulphinpyrazone_controls``` were controls

- ```r self_ult_drug_tally$colchicine_cases``` people taking cochicine, ```r self_ult_drug_tally$colchicine_controls``` were controls



```{r self_ult_gwas_sex_plot2, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_self_ult) %>% 
  filter(!is.na(goutaff), fam_self_ult == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Self Report and ULT Sex Distribution")
```

```{r self_ult_gwas_bmi_plot2, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_self_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Self Report and ULT Gout Criteria")
```

BMI Summary:
  
```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
                select(fam_self_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
                filter(fam_self_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
                mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
                group_by(cc) %>% 
                summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Self report and ULT summary table

```{r self_ult_summary_table2, echo = FALSE, results = 'asis'}
pandoc.table(self_ult_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self_ult'
self_ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(self_ult_kot[['sum_table']]) <- NULL
pandoc.table(self_ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```


### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self_ult'
self_ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(self_ult_kot[['sum_table']]) <- NULL
pandoc.table(self_ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout self and ult - adjusted: age sex waist and waist to height](controls_no_diuretics_self_ult_adjusted.png)

\newpage
## Combined tables

\newpage
\blandscape



```{r, echo = FALSE, results = 'asis'}
kot_t1 <- read.csv("kottgen_table1.csv", header = TRUE, stringsAsFactors = FALSE)
kot_t1_gout <- kot_t1[,c("SNP","CHR","BP","grail_gene","OR","gout_pvalue", "A1", "A2")]
names(kot_t1_gout) <- c("SNP","CHR","BP","grail_gene","OR","P", "A1","A2")
kot_t1_gout$lower95 <- NA
kot_t1_gout$upper95 <- NA
kot_t1_gout$SE <- NA
```

```{r, echo = FALSE, message=FALSE}
library(RMySQL)
library(data.table)
library(dplyr)
drv = dbDriver("MySQL")
db = dbConnect(drv, default.file = '~/.my.cnf', dbname="merrimanDW_test")

ukvariant <- dbGetQuery(db, paste('select * from dimVariant where snp in (', paste(paste0("'",kot_t1$SNP,"'"), collapse = ','),")"))
```



```{r, echo=FALSE, results = 'asis'}
kottgen_table1 <- list(all =merge(all_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), winnard = merge(winnard_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), hosp = merge(hosp_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), self = merge(self_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), ult = merge(ult_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), self_ult = merge(self_ult_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), kottgen_gout = kot_t1_gout) 
for (name in names(kottgen_table1)){
  kottgen_table1[[name]]$gout <-  name
}



kt1 <- do.call(rbind, kottgen_table1)
kt1 <-kt1[order(kt1$SNP), ! colnames(kt1) %in% c('sort')]
row.names(kt1) <- NULL
#pandoc.table(kt1[,c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P","gout")], caption = 'Combined Kottgen Table 1 All Gout Classifications' )
pandoc.table(kt1[,c('SNP','CHR','BP',"A1","A2", "grail_gene","OR","P","gout")], caption = 'Combined Kottgen Table 1 All Gout Classifications - adjusted' )
```

\elandscape

\newpage

```{r, echo = FALSE, fig.align='centre', fig.width= 8, fig.height=8}
ggplot(data = kt1, aes(x = gout, y = OR)) + geom_point() + facet_wrap(~SNP, scales = 'free_y') + geom_linerange(aes(x = gout, ymin = lower95, ymax = upper95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90)) 

```


```{r, echo = FALSE, fig.width=8, fig.height=8, eval = FALSE}
dat_kottgen_table1 <- list(all =all_kot[['dat']], winnard = winnard_kot[['dat']], hosp = hosp_kot[['dat']], self = self_kot[['dat']], ult = ult_kot[['dat']],self_ult = self_ult_kot[['dat']]) 
for (name in names(dat_kottgen_table1)){
  dat_kottgen_table1[[name]]$gout <-  name
}
dat_kt1 <- do.call(rbind, dat_kottgen_table1)
dat_kt1 <-dat_kt1[order(dat_kt1$SNP), ! colnames(dat_kt1) %in% c('sort')]
row.names(dat_kt1) <- NULL
#pandoc.table(dat_kt1[,], caption = 'Combined Kottgen Table 1 All Gout Classifications' )

ggplot(data = dat_kt1, aes(x = gout, y = OR)) + geom_point() + facet_wrap(~SNP, scales = 'free_y') + geom_linerange(aes(x = gout, ymin = L95, ymax = L95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90)) + stat_summary(fun.y=mean, geom="point", shape=18,size=3, color="red") 
``` 


\newpage

# Genotyped - Controls only diuretics 





```{r load data3,echo = FALSE, message=FALSE, warning=FALSE}
control_cond <- 'controls_diuretics'

load(paste0(data_dir,"ukbiobank_genotyped2016-04-26.RData"))
genotyped$waist_height_ratio <- genotyped$f.48.0.0 / genotyped$f.50.0.0

source("create_kottgen_table1.R")
source('read_fams.R')

gout_cond <- 'all'
all_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(all_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_all <- NA
genotyped[genotyped$f.eid %in% all_fam$IID & 
                  genotyped$f.eid %in% all_fam[which(all_fam$AFF==1 | all_fam$AFF == 2),"IID"], "fam_all"] <-1

genotyped$fam <- NA
genotyped[genotyped$f.eid %in% all_fam$IID,"fam"] <- 1

gout_cond <- 'self'
self_fam  <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(self_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_self <- NA
genotyped[genotyped$f.eid %in% self_fam$IID & 
                  genotyped$f.eid %in% self_fam[which(self_fam$AFF==1 | self_fam$AFF == 2),"IID"], "fam_self"] <-1

gout_cond <- 'self_ult'
self_ult_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(self_ult_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_self_ult <- NA
genotyped[genotyped$f.eid %in% self_ult_fam$IID & 
                  genotyped$f.eid %in% self_ult_fam[which(self_ult_fam$AFF==1 | self_ult_fam$AFF == 2),"IID"], "fam_self_ult"] <-1

gout_cond <- 'winnard'
winnard_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(winnard_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_winnard <- NA
genotyped[genotyped$f.eid %in% winnard_fam$IID & 
                  genotyped$f.eid %in% winnard_fam[which(winnard_fam$AFF==1 | winnard_fam$AFF == 2),"IID"], "fam_winnard"] <-1

gout_cond <- 'ult'
ult_fam <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(ult_fam) <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_ult <- NA
genotyped[genotyped$f.eid %in% ult_fam$IID & 
                  genotyped$f.eid %in% ult_fam[which(ult_fam$AFF==1 | ult_fam$AFF == 2),"IID"], "fam_ult"] <-1


gout_cond <- 'hosp'
hosp_fam  <- read.table(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond), header=FALSE)
colnames(hosp_fam)  <- c("FID","IID","PID","MID", "SEX","AFF")
genotyped$fam_hosp <- NA
genotyped[genotyped$f.eid %in% hosp_fam$IID & 
                  genotyped$f.eid %in% hosp_fam[which(hosp_fam$AFF==1 | hosp_fam$AFF == 2),"IID"], "fam_hosp"] <-1
```



There are a total of ```r as.character(genotyped %>% select(fam)  %>% filter(fam == 1) %>% tally())``` people with genotypes

The GWAS were limited to only British, Irish, and "any other white background" ethnicities because there were only ```r as.data.frame(genotyped  %>% filter(!is.na(goutaff) & !(f.21000.0.0 %in% c(1001,1002,1003)) & f.eid %in% all_fam$IID)  %>%  select(f.21000.0.0, goutaff)  %>% group_by(goutaff, f.21000.0.0)  %>%  tally()  %>% summarise(sum(n)))[2,2]``` gout cases from all other ethnicities with genotypes

## All


```{r all_gwas_stats3, echo=FALSE}
gout_cond <- 'all'
all <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
all_drug_tally <- all[[2]]
all_summary_table <-  all[[1]]

```
For the GWAS based on all gout criteria there were ```r all_summary_table['n',"Gout"]``` cases and ```r all_summary_table['n',"Control"]``` controls

There were:
  
- ```r as.character(all_drug_tally$diuretics_cases)``` cases on diuretics, ```r all_drug_tally$diuretics_controls``` were controls

- ```r all_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r all_drug_tally$allopurinol_controls``` were controls

- ```r all_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r all_drug_tally$febuxostat_controls``` were controls

- ```r all_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r all_drug_tally$sulphinpyrazone_controls``` were controls

- ```r all_drug_tally$colchicine_cases``` people taking cochicine, ```r all_drug_tally$colchicine_controls``` were controls

Age summary

```{r all_age_summary3, echo = FALSE}
as.data.frame(genotyped  %>% 
  select(fam_all, goutaff, f.21003.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21003.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21003.0.0), sd = sd(f.21003.0.0), min = min(f.21003.0.0),median = median(f.21003.0.0), max = max(f.21003.0.0) ), row.names = NULL)
```

Sex summary

```{r all_gwas_sex_plot3, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_all) %>% 
  filter(!is.na(goutaff), fam_all == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("All Sex Distribution")
```

```{r all_gwas_bmi_plot3, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_all, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("All Gout Criteria")
```

```{r all_gwas_wh_plot3, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_all, goutaff, waist_height_ratio , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(waist_height_ratio)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = waist_height_ratio)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("Waist to Height Ratio") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("All Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval = FALSE}
as.data.frame(genotyped  %>% 
  select(fam_all, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_all == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```


### All summary table



```{r all_summary_table3, echo = FALSE, results= 'asis'}
pandoc.table(all_summary_table, caption = 'Summary Table - All')
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results='asis'}
gout_cond <- 'all'
all_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(all_kot[['sum_table']]) <- NULL
pandoc.table(as.data.frame(all_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")], row.names = NULL), caption = 'Kottgen table 1 SNPs - All')

```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results='asis'}
gout_cond <- 'all'
all_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(all_kot[['sum_table']]) <- NULL
pandoc.table(as.data.frame(all_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")], row.names = NULL), caption = 'Kottgen table 1 SNPs - All')

```

![Gout all - adjusted: age sex waist and waist to height](controls_diuretics_all_adjusted.png)

\pagebreak

\newpage
## Winnard

```{r winnard_gwas_stats3, echo = FALSE}
gout_cond <- 'winnard'
winnard <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
winnard_drug_tally <- winnard[[2]]
winnard_summary_table <-  winnard[[1]]

```


For the GWAS based on the Winnard et al gout criteria there were ```r winnard_summary_table['n',"Gout"]``` cases and ```r as.character(winnard_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(winnard_drug_tally$diuretics_cases)``` cases on diuretics, ```r winnard_drug_tally$diuretics_controls``` were controls

- ```r winnard_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r winnard_drug_tally$allopurinol_controls``` were controls

- ```r winnard_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r winnard_drug_tally$febuxostat_controls``` were controls

- ```r winnard_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r winnard_drug_tally$sulphinpyrazone_controls``` were controls

- ```r winnard_drug_tally$colchicine_cases``` people taking cochicine, ```r winnard_drug_tally$colchicine_controls``` were controls


```{r winnard_gwas_sex_plot3, echo = FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_winnard) %>% 
  filter(!is.na(goutaff), fam_winnard == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Winnard Sex Distribution")
```

```{r winnard_gwas_bmi_plot3, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_winnard, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_winnard == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Winnard Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_winnard, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_winnard == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc,sex) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```


### Winnard summary table

```{r winnard_summary_table3, echo = FALSE, results = 'asis'}
pandoc.table(winnard_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'winnard'
winnard_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(winnard_kot[['sum_table']]) <- NULL
pandoc.table(winnard_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'winnard'
winnard_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(winnard_kot[['sum_table']]) <- NULL
pandoc.table(winnard_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

![Gout winnard - adjusted: age sex waist and waist to height](controls_diuretics_winnard_adjusted.png)

\pagebreak

\newpage
## Hospital

```{r hosp_gwas_stats3, echo = FALSE}
gout_cond <- 'hosp'
hosp <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
hosp_drug_tally <- hosp[[2]]
hosp_summary_table <-  hosp[[1]]

```


For the GWAS based on hospital diagnosis of gout criteria there were ```r hosp_summary_table['n',"Gout"]``` cases and ```r as.character(hosp_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(hosp_drug_tally$diuretics_cases)``` cases on diuretics, ```r hosp_drug_tally$diuretics_controls``` were controls

- ```r hosp_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r hosp_drug_tally$allopurinol_controls``` were controls

- ```r hosp_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r hosp_drug_tally$febuxostat_controls``` were controls

- ```r hosp_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r hosp_drug_tally$sulphinpyrazone_controls``` were controls

- ```r hosp_drug_tally$colchicine_cases``` people taking cochicine, ```r hosp_drug_tally$colchicine_controls``` were controls


```{r hosp_gwas_sex_plot3, echo = FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_hosp) %>% 
  filter(!is.na(goutaff), fam_hosp == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Hospital Sex Distribution")
```

```{r hosp_gwas_bmi_plot3, echo=FALSE, message=FALSE, warning=FALSE, eval = FALSE}
genotyped  %>% 
  select(fam_hosp, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_hosp == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Hospital Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_hosp, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_hosp == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Hospital summary table

```{r hosp_summary_table3, echo = FALSE, results= 'asis'}
pandoc.table(hosp_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'hosp'
hosp_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(hosp_kot[['sum_table']]) <- NULL
pandoc.table(hosp_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'hosp'
hosp_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(hosp_kot[['sum_table']]) <- NULL
pandoc.table(hosp_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])

```

![Gout hospital - adjusted: age sex waist and waist to height](controls_diuretics_hosp_adjusted.png)

\pagebreak

\newpage
## Self Reported

```{r self_gwas_stats3, echo=FALSE}
gout_cond <- 'self'
self <-  read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
self_drug_tally <- self[[2]]
self_summary_table <-  self[[1]]

```



For the GWAS based on self reported gout criteria there were ```r self_summary_table['n',"Gout"]``` cases and ```r as.character(self_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(self_drug_tally$diuretics_cases)``` cases on diuretics, ```r self_drug_tally$diuretics_controls``` were controls

- ```r self_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r self_drug_tally$allopurinol_controls``` were controls

- ```r self_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r self_drug_tally$febuxostat_controls``` were controls

- ```r self_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r self_drug_tally$sulphinpyrazone_controls``` were controls

- ```r self_drug_tally$colchicine_cases``` people taking cochicine, ```r self_drug_tally$colchicine_controls``` were controls



```{r self_gwas_sex_plot3, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_self) %>% 
  filter(!is.na(goutaff), fam_self == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Self Report Sex Distribution")
```

```{r self_gwas_bmi_plot3, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_self, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Self Report Gout Criteria")
```

BMI Summary:

```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
  select(fam_self, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc) %>% 
  summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Self report summary table

```{r self_summary_table3, echo = FALSE, results= 'asis'}
pandoc.table(self_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self'
self_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(self_kot[['sum_table']]) <- NULL
pandoc.table(self_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self'
self_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(self_kot[['sum_table']]) <- NULL
pandoc.table(self_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout self - adjusted: age sex waist and waist to height](controls_diuretics_self_adjusted.png)

\pagebreak

\newpage
## ULT
```{r ult_gwas_stats3, echo = FALSE}
gout_cond <- 'ult'
ult <- read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
ult_drug_tally <- ult[[2]]
ult_summary_table <-  ult[[1]]
```

For the GWAS based on ULT gout criteria there were ```r ult_summary_table['n',"Gout"]``` cases and ```r as.character(ult_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(ult_drug_tally$diuretics_cases)``` cases on diuretics, ```r ult_drug_tally$diuretics_controls``` were controls

- ```r ult_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r ult_drug_tally$allopurinol_controls``` were controls

- ```r ult_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r ult_drug_tally$febuxostat_controls``` were controls

- ```r ult_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r ult_drug_tally$sulphinpyrazone_controls``` were controls

- ```r ult_drug_tally$colchicine_cases``` people taking cochicine, ```r ult_drug_tally$colchicine_controls``` were controls



```{r ult_gwas_sex_plot3, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_ult) %>% 
  filter(!is.na(goutaff), fam_ult == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("ULT Sex Distribution")
```

```{r ult_gwas_bmi_plot3, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("ULT Gout Criteria")
```

BMI Summary:
  
```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
                select(fam_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
                filter(fam_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
                mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
                group_by(cc) %>% 
                summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### ULT summary table

```{r ult_summary_table3, echo = FALSE, results = 'asis'}
pandoc.table(ult_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'ult'
ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(ult_kot[['sum_table']]) <- NULL
pandoc.table(ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'ult'
ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(ult_kot[['sum_table']]) <- NULL
pandoc.table(ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout ult - adjusted: age sex waist and waist to height](controls_diuretics_ult_adjusted.png)

\newpage
## Self Reported and ULT

```{r self_ult_gwas_stats3, echo = FALSE}
gout_cond <- 'self_ult'
self_ult <- read_fams(paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/',"chrallimpv1.fam_",gout_cond))
self_ult_drug_tally <- self_ult[[2]]
self_ult_summary_table <-  self_ult[[1]]
```

For the GWAS based on self report and ULT gout criteria there were ```r self_ult_summary_table['n',"Gout"]``` cases and ```r as.character(self_ult_summary_table['n',"Control"])``` controls

There were:
  
- ```r as.character(self_ult_drug_tally$diuretics_cases)``` cases on diuretics, ```r self_ult_drug_tally$diuretics_controls``` were controls

- ```r self_ult_drug_tally$allopurinol_cases``` cases taking allopurinol, ```r self_ult_drug_tally$allopurinol_controls``` were controls

- ```r self_ult_drug_tally$febuxostat_cases``` cases taking febuxostat, ```r self_ult_drug_tally$febuxostat_controls``` were controls

- ```r self_ult_drug_tally$sulphinpyrazone_cases``` cases taking sulphinpyrazone, ```r self_ult_drug_tally$sulphinpyrazone_controls``` were controls

- ```r self_ult_drug_tally$colchicine_cases``` people taking cochicine, ```r self_ult_drug_tally$colchicine_controls``` were controls



```{r self_ult_gwas_sex_plot3, echo = FALSE, eval=FALSE}
genotyped  %>% 
  select(goutaff, f.31.0.0, fam_self_ult) %>% 
  filter(!is.na(goutaff), fam_self_ult == 1 ) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)) , cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  group_by(cc, sex)  %>% tally()  %>% 
  ggplot(., aes(x = sex, y = n)) + 
  geom_bar(aes(fill =sex), stat = 'identity') +
  geom_text(aes(label=n), vjust=-0.3, color="black",
            position = position_dodge(0.9), size=3.5)+
  facet_wrap(~cc) +
  theme_bw() + 
  theme(legend.position = "none") + 
  ggtitle("Self Report and ULT Sex Distribution")
```

```{r self_ult_gwas_bmi_plot3, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
genotyped  %>% 
  select(fam_self_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
  filter(fam_self_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
  mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
  ggplot(., aes(x = cc, y = f.21001.0.0)) + 
  geom_boxplot() + xlab("Gout Affection") + 
  ylab("BMI") + 
  theme_bw() + 
  facet_wrap(~sex) +
  ggtitle("Self Report and ULT Gout Criteria")
```

BMI Summary:
  
```{r, echo = FALSE, eval=FALSE}
as.data.frame(genotyped  %>% 
                select(fam_self_ult, goutaff, f.21001.0.0 , f.31.0.0) %>% 
                filter(fam_self_ult == 1, !is.na(goutaff), !is.na(f.21001.0.0)) %>% 
                mutate(sex = ifelse(f.31.0.0 == 0, "Female", ifelse(f.31.0.0 == 1, "Male",NA)), cc = ifelse(goutaff == 1, "Case", "Control")) %>% 
                group_by(cc) %>% 
                summarise(mean = mean(f.21001.0.0), sd = sd(f.21001.0.0), min = min(f.21001.0.0),median = median(f.21001.0.0), max = max(f.21001.0.0) ), row.names = NULL)
```

### Self report and ULT summary table

```{r self_ult_summary_table3, echo = FALSE, results = 'asis'}
pandoc.table(self_ult_summary_table)
```

Sex as % male  
Age and BMI given as mean +/- sd  
Ethnicity in order British, Irish, and Any other white background

### Kottgen table 1 SNPs - unadjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self_ult'
self_ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/unadjusted/kottgen_results.txt'))
row.names(self_ult_kot[['sum_table']]) <- NULL
pandoc.table(self_ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

### Kottgen table 1 SNPs - age sex waist and waist to height ratio adjusted

```{r, echo = FALSE, results = 'asis'}
gout_cond <- 'self_ult'
self_ult_kot <- create_kottgen_table1(kottgen_loci =paste0(scratch_dir,"GWAS_all_controls/",control_cond,'/',gout_cond,'/adjusted/kottgen_results.txt'))
row.names(self_ult_kot[['sum_table']]) <- NULL
pandoc.table(self_ult_kot[['sum_table']][c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P")])
```

![Gout self and ult - adjusted: age sex waist and waist to height](controls_diuretics_self_ult_adjusted.png)

\newpage
## Combined Tables

\newpage
\blandscape



```{r, echo = FALSE, results = 'asis'}
kot_t1 <- read.csv("kottgen_table1.csv", header = TRUE, stringsAsFactors = FALSE)
kot_t1_gout <- kot_t1[,c("SNP","CHR","BP","grail_gene","OR","gout_pvalue", "A1", "A2")]
names(kot_t1_gout) <- c("SNP","CHR","BP","grail_gene","OR","P", "A1","A2")
kot_t1_gout$lower95 <- NA
kot_t1_gout$upper95 <- NA
kot_t1_gout$SE <- NA
```

```{r, echo = FALSE, message=FALSE}
library(RMySQL)
library(data.table)
library(dplyr)
drv = dbDriver("MySQL")
db = dbConnect(drv, default.file = '~/.my.cnf', dbname="merrimanDW_test")

ukvariant <- dbGetQuery(db, paste('select * from dimVariant where snp in (', paste(paste0("'",kot_t1$SNP,"'"), collapse = ','),")"))
```



```{r, echo=FALSE, results = 'asis'}
kottgen_table1 <- list(all =merge(all_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), winnard = merge(winnard_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), hosp = merge(hosp_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), self = merge(self_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), ult = merge(ult_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), self_ult = merge(self_ult_kot[['sum_table']], ukvariant[,c("snp","A1","A2")],by.x = "SNP", by.y = "snp", all.x = TRUE), kottgen_gout = kot_t1_gout) 
for (name in names(kottgen_table1)){
  kottgen_table1[[name]]$gout <-  name
}



kt1 <- do.call(rbind, kottgen_table1)
kt1 <-kt1[order(kt1$SNP), ! colnames(kt1) %in% c('sort')]
row.names(kt1) <- NULL
#pandoc.table(kt1[,c('SNP','CHR','BP', "grail_gene","OR","lower95","upper95","P","gout")], caption = 'Combined Kottgen Table 1 All Gout Classifications' )
pandoc.table(kt1[,c('SNP','CHR','BP',"A1","A2", "grail_gene","OR","P","gout")], caption = 'Combined Kottgen Table 1 All Gout Classifications - adjusted' )
```

\elandscape

\newpage

```{r, echo = FALSE, fig.align='centre', fig.width= 8, fig.height=8}
ggplot(data = kt1, aes(x = gout, y = OR)) + geom_point() + facet_wrap(~SNP, scales = 'free_y') + geom_linerange(aes(x = gout, ymin = lower95, ymax = upper95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90)) 

```


```{r, echo = FALSE, fig.width=8, fig.height=8, eval = FALSE}
dat_kottgen_table1 <- list(all =all_kot[['dat']], winnard = winnard_kot[['dat']], hosp = hosp_kot[['dat']], self = self_kot[['dat']], ult = ult_kot[['dat']], self_ult = self_ult_kot[['dat']]) 
for (name in names(dat_kottgen_table1)){
  dat_kottgen_table1[[name]]$gout <-  name
}
dat_kt1 <- do.call(rbind, dat_kottgen_table1)
dat_kt1 <-dat_kt1[order(dat_kt1$SNP), ! colnames(dat_kt1) %in% c('sort')]
row.names(dat_kt1) <- NULL
#pandoc.table(dat_kt1[,], caption = 'Combined Kottgen Table 1 All Gout Classifications' )

ggplot(data = dat_kt1, aes(x = gout, y = OR)) + geom_point() + facet_wrap(~SNP, scales = 'free_y') + geom_linerange(aes(x = gout, ymin = L95, ymax = L95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90)) + stat_summary(fun.y=mean, geom="point", shape=18,size=3, color="red") 
``` 


